test/controllers/AuthorizedToViewControllerSpec.scala

```diff
@@ -498,8 +498,33 @@ class AuthorizedToViewControllerSpec extends SpecBase {
       }
     }
 
+    "Display error message if searching your own XI EORI number" in new Setup {
+      val gbAuthCsvFiles: Seq[StandingAuthorityFile] = Seq(gbStandingAuth1, gbStandingAuth2)
+      when(mockSdesConnector.getAuthoritiesCsvFiles(any)(any)).thenReturn(Future.successful(gbAuthCsvFiles))
+
+      when(mockDataStoreService.getXiEori(any)(any)).thenReturn(Future.successful(Some("XI123456789912")))
+
+      running(app) {
+        val request = fakeRequest(POST,
+          routes.AuthorizedToViewController.onSubmit().url).withFormUrlEncodedBody(
+          "value" -> "XI123456789912")
+
+        val result = route(app, request).value
+        val html = Jsoup.parse(contentAsString(result))
+        status(result) shouldBe BAD_REQUEST
+
+        html.getElementById("gb-csv-authority-link").html() mustBe
+          messages(app)("cf.authorities.notification-panel.a.gb-authority")
+        html.getElementById("gb-csv-authority-link").attr("href") mustBe gbStanAuthFile154Url
+
+        html.text().contains("You cannot search your own EORI number") shouldBe true
+      }
+    }
+
     "Display error message if searching your own account number" in new Setup {
       when(mockDataStoreService.getXiEori(any)(any)).thenReturn(Future.successful(None))
+      when(mockApiService.getAccounts(ArgumentMatchers.eq(newUser().eori))(any)).thenReturn(Future.successful(cdsAccounts))
```

Comment:
Could you please split (better to do after thenReturn) the line over multiple line as its very long line? Same comment for line no 527, 543, 544, 641 and 642